front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -213,50 +251,93 @@ const DocumentUploadOrEditModal = ({
   };
 
   const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
-    const selectedFile = e.target.files?.[0];
-    if (!selectedFile) {
+    // Enforce single file upload
+    const files = e.target.files;
+    if (files && files.length > 1) {
+      sendNotification({
+        type: "error",
+        title: "Multiple files",
+        description: "Please upload only one file at a time.",
+      });
       return;
     }
 
-    setUploading(true);
     try {
-      if (selectedFile.size > MAX_FILE_SIZES.plainText) {
-        sendNotification({
-          type: "error",
-          title: "File too large",
-          description: `Please upload a file smaller than ${maxFileSizeToHumanReadable(MAX_FILE_SIZES.plainText)}.`,
-        });
-        setUploading(false);
+      // Create a file -> Allows to get processed text content from Tika via the file API.
+      const selectedFile = files?.[0];
+      if (!selectedFile) {
         return;
       }
-
-      const res = await handleFileUploadToText(selectedFile);
-      if (res.isErr()) {
-        return new Err(res.error);
+      const fileBlobs = await fileUploaderService.handleFilesUpload([
+        selectedFile,
+      ]);
+      if (!fileBlobs || fileBlobs.length == 0 || !fileBlobs[0].fileId) {
+        fileUploaderService.resetUpload();
+        return new Err(
+          new Error(
+            "Error uploading file. Please try again or contact support."
+          )
+        );
       }
-      setDocumentState((prev) => ({ ...prev, text: res.value.content }));
+
+      // triggers content extraction -> documentState.text update
+      setFileId(fileBlobs[0].fileId);
     } catch (error) {
       sendNotification({
         type: "error",
         title: "Error uploading file",
         description: error instanceof Error ? error.message : String(error),
       });
-    } finally {
-      setUploading(false);
     }
   };
 
+  // Effect: Set the document state when the document is loaded
+  useEffect(() => {
+    if (!initialId) {
+      setDocumentState({
+        name: "",
+        text: "",
+        tags: [],
+        sourceUrl: "",
+      });
+    } else if (document && isCoreAPIDocumentType(document)) {
+      setDocumentState((prev) => ({
+        ...prev,
+        name: initialId,
+        text: document.text ?? "",
+        tags: document.tags,
+        sourceUrl: document.source_url ?? "",
+      }));
+    }
+  }, [initialId, document]);
+
+  // Effect: Validate the document state
+  useEffect(() => {
```

Comment:
I can't just move it up because the above effect (295) is only ran once when document loads. The other effect is ran every time the inputs for content or name change. 
I could also setIsValidDocument in the above effect as a shortcut, but that would require creating a function to not duplicate the logic of whether a document is valid or not, and call it twice.-> Felt overkill for something that already works.